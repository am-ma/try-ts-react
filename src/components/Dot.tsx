import React, { CSSProperties } from 'react';
import DotValues from '~/models/DotValues.model';

interface DotProps {
  values: DotValues;
  onClick: Function;
}

interface DotState {}

export default class Dot extends React.Component<DotProps, DotState> {
  renderStyle() {
    if (!this.props.values.isFill) {
      return {} as CSSProperties;
    }

    return {
      backgroundColor: this.props.values.color,
    } as CSSProperties;
  }

  render() {
    return <button style={this.renderStyle()} className="dot" onClick={() => this.props.onClick()}></button>;
  }
}
