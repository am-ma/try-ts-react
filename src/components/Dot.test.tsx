import React from 'react';
import Dot from './Dot';
import renderer from 'react-test-renderer';
import DotValues from '~/models/DotValues.model';

describe('Dot', () => {
  describe('render', () => {
    it('will render when isFill is TRUE', () => {
      const onClick = jest.fn();
      const values: DotValues = {
        isFill: true,
        color: '#000',
      };
      const result = renderer.create(<Dot values={values} onClick={onClick} />).toJSON();

      expect(result).toMatchSnapshot();
    });
    it('will render when isFill is FALSE', () => {
      const onClick = jest.fn();
      const values: DotValues = {
        isFill: false,
        color: '#000',
      };
      const result = renderer.create(<Dot values={values} onClick={onClick} />).toJSON();

      expect(result).toMatchSnapshot();
    });
  });
});
