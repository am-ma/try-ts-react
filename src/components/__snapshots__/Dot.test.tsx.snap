// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot render will render when isFill is FALSE 1`] = `
<button
  className="dot"
  onClick={[Function]}
  style={Object {}}
/>
`;

exports[`Dot render will render when isFill is TRUE 1`] = `
<button
  className="dot"
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "#000",
    }
  }
/>
`;
